### PR TITLE
DNM: Adding support for nvapi

### PR DESCRIPTION
**Pull request by samcake**
THis PR adds support for nvapi to get better info specifically for when running on a nvidia GPU.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 99032dd1201a32a2701b1c087e8cfb11a60d932f available at: https://gist.github.com/71834615ad2736a643d87b723016eb23<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-99032dd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-99032dd.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c3a867ff733559083e0a1f05d581e24985e050f6 available at: https://gist.github.com/71834615ad2736a643d87b723016eb23<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-c3a867f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-c3a867f.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a9beb1d6337d3d19b8ac256e5e93b34d1d1a4090 available at: https://gist.github.com/71834615ad2736a643d87b723016eb23<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-a9beb1d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-a9beb1d.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 020ecb3c4b49fccc9de7f23e1528d5d879a21801 available at: https://gist.github.com/71834615ad2736a643d87b723016eb23<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-020ecb3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-020ecb3.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8c870acf4914bba25177354ce02f8495480d287f available at: https://gist.github.com/71834615ad2736a643d87b723016eb23<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-8c870ac.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-8c870ac.dmg

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 1]
```diff
@@ -59,9 +59,9 @@ EntityTreeRenderer::EntityTreeRenderer(bool wantScripts, AbstractViewStateInterf
     REGISTER_ENTITY_TYPE_WITH_FACTORY(Light, RenderableLightEntityItem::factory)
     REGISTER_ENTITY_TYPE_WITH_FACTORY(Text, RenderableTextEntityItem::factory)
     // Offscreen web surfaces are incompatible with nSight
-    if (!nsightActive()) {
+  //  if (!nsightActive()) {
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: typo?

**Comment by ZappoMan** [comment]
ran it on my environment...
before running

```
[10/22 21:56:13] [DEBUG] [hifi.gpu.gl] Nvidia is in the house: 
[10/22 21:56:13] [DEBUG] [hifi.gpu.gl] AvailableDedicatedVideoMemory [MB] :  3977 
[10/22 21:56:13] [DEBUG] [hifi.gpu.gl] DedicatedVideoMemory [MB] :  4096 
[10/22 21:56:13] [DEBUG] [hifi.gpu.gl] CurrentAvailableDedicatedVideoMemory [MB] :  3136
```

Before running.... 
**GPU Dedicated Bytes Total:**  527,084 KB (514MB? or 527MB?)
**GPU Committed Bytes Total:** 755,664 KB (738MB? or 755MB?)

Neither of those values total up to the "available" vs the current "available" -- but committed bytes gets closest.

My gut says that using that **CurrentAvailableDedicatedVideoMemory** as the baseline for us to work off would maybe be a good idea... instead of the max value... @jherico @samcake - what do you think?

**Comment by hifi-gustavo** [comment]
Build console output for commit: e1906bde354269f3dd0d62f48df36e7d0368fce3 available at: https://gist.github.com/71834615ad2736a643d87b723016eb23<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-e1906bd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-e1906bd.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0ed955292218b54648c1ec6b09a2cdbe98743c77 available at: https://gist.github.com/71834615ad2736a643d87b723016eb23<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-0ed9552.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8875-0ed9552.dmg

**Pull request closed without merging**